### PR TITLE
Refactoring, StateObserver and Python Tooling

### PR DESCRIPTION
**Pull request by devandt**
Important files: 
- [observer.py](src/gym_electric_motor/observers/observer.py)
- [DEVELOPMENT.md](DEVELOPMENT.md) 
- many small changes

# [2.0.1] - Unreleased
## Added
- Ruff: Python linter & formatter (see [DEVELOPMENT.md](DEVELOPMENT.md))
- StateObserver: An easy way to get state values with error checking [example](examples/observers/state_observer_example.py)
- Integrated gem_controls repository into gem. classic_controllers will be removed in further version
- Using pyproject.toml, dropping deprecated setup.py
- Enabled Gymnasium env checker [see here](https://gymnasium.farama.org/api/experimental/wrappers/#gymnasium.experimental.wrappers.PassiveEnvCheckerV0)
## Changed
- Linted and formatted all files
- Changed max. steps in some test files to improve test speed by 30%
## Fixed

### CODE REVIEW

File: src/gym_electric_motor/physical_systems/electric_motors/dc_series_motor.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    _default_motor_parameter = {
+        "r_a": 16e-3,
+        "r_e": 48e-3,
+        "l_a": 19e-6,
+        "l_e_prime": 1.7e-3,
+        "l_e": 5.4e-3,
+        "j_rotor": 0.0025,
+    }
+    _default_nominal_values = dict(omega=300, torque=16.0, i=97, i_a=97, i_e=97, u=60, u_a=60, u_e=60)
+    _default_limits = dict(omega=400, torque=38.0, i=210, i_a=210, i_e=210, u=60, u_a=60, u_e=60)
```
Comment (max-schenke) [inline src/gym_electric_motor/physical_systems/electric_motors/dc_series_motor.py:60]: Should one specify all these values as floats by putting ".0" behind them?

(Applies to several files)

File: src/gym_electric_motor/physical_systems/electric_motors/externally_excited_synchronous_motor.py [hunk 2]
```diff
@@ -92,48 +93,83 @@ class ExternallyExcitedSynchronousMotor(SynchronousMotor):
...
+                    0,
+                    -mp["r_e"] / sigma,
+                    -mp["l_m"] / (sigma * mp["l_d"]),
+                    0,
+                    1 / sigma,
+                    0,
+                    -mp["p"] * mp["l_m"] * mp["l_q"] / (sigma * mp["l_d"]),
+                    0,
+                ],
+                [mp["p"], 0, 0, 0, 0, 0, 0, 0, 0, 0],
```
Comment (max-schenke) [inline src/gym_electric_motor/physical_systems/electric_motors/externally_excited_synchronous_motor.py:166]: On the one hand, I get that the formatter prefers this shape of code.
On the other hand, the matrix calculations involved have been more intuitive the way it was before refactoring. 
I guess this needs further discussion.

(Applies to several files)

Comment (devandt) [reply src/gym_electric_motor/physical_systems/electric_motors/externally_excited_synchronous_motor.py:166]: Yea, formating should definitely be disabled for this kind of calucations.
Leaving this here for future reference:

```
# fmt: off

def my_unformatted_function():
   do_something()

# fmt: on
```

File: src/gym_electric_motor/physical_systems/electric_motors/three_phase_motor.py [hunk 3]
```diff
@@ -1,28 +1,23 @@
...
+    This includes the system equations, the motor parameters of the equivalent circuit diagram,
+    as well as limits and bandwidth.
     """
+
     # transformation matrix from abc to alpha-beta representation
-    _t23 = 2 / 3 * np.array([
-        [1, -0.5, -0.5],
-        [0, 0.5 * np.sqrt(3), -0.5 * np.sqrt(3)]
-    ])
+    _t23 = 2 / 3 * np.array([[1, -0.5, -0.5], [0, 0.5 * np.sqrt(3), -0.5 * np.sqrt(3)]])
```
Comment (max-schenke) [inline src/gym_electric_motor/physical_systems/electric_motors/three_phase_motor.py:17]: Same issue. The formatter made the code less understandable (in my opinion).

**Review by max-schenke — COMMENTED** [review]
No further objections apart from my earlier minor remarks.

An execution test will follow.

**Comment by max-schenke** [comment]
I was not successful to run the unit tests locally:
`ModuleNotFoundError: No module named 'classic_controllers'`

Did I miss an optional install or is something missing from the requirements?
Also, GitHub still reports the tests to not be successful for Python 3.8. Is that fixable?

**Comment by max-schenke** [comment]
Integration test still fails on my machine, all other tests seems successful. 

Also, GitHub still reports Version 3.8 to have failing tests.

Formatting issues persist and make it hard to read coded matrices.

**Pull request merged**